Terminal.Gui/ConsoleDrivers/NetDriver.cs

```diff
@@ -196,7 +197,14 @@ void GetConsoleKey ()
 						break;
 					}
 				} else if (consoleKeyInfo.KeyChar == (char)Key.Esc && isEscSeq) {
-					DecodeEscSeq (ref newConsoleKeyInfo, ref key, cki, ref mod);
+					if (cki == null) {
+						// This is an error. We got an ESC while we're in an ESC sequence already
+						// but there's no record of the ESC sequence.
+						// Ignore this for now until we can figure out how we got in this state
+						isEscSeq = false;
+					} else {
+						DecodeEscSeq (ref newConsoleKeyInfo, ref key, cki, ref mod);
+					}
 					cki = null;
 					break;
```

Comment:
Instead this code I think is better do the follow:

```cs
					DecodeEscSeq (ref newConsoleKeyInfo, ref key, cki, ref mod);
					cki = null;
					if (!Console.KeyAvailable) {
						isEscSeq = false;
					}
					break;
```
So, it process the Esc and if there isn't `Console.KeyAvailable` we've certain that isn't a escape sequence but just a simple Esc and set the `isEscSeq` to false. `cki` is always set to null and if it's a escape sequence the prior condition will add the Esc key to `cki`.

Reply:
I think you're right this is slightly more robust than my fix. I'll change to use your suggestion. Thanks!

However, I don't think it's correct to say "if there isn't `Console.KeyAvailable` we've certain that isn't a escape sequence". It is PROBABLY true with Windows Terminal, but it is POSSIBLE that another terminal is sending keys slowly enough that when this line of code executes the next key simply hasn't arrived yet.

Reply:
> However, I don't think it's correct to say "if there isn't `Console.KeyAvailable` we've certain that isn't a escape sequence". It is PROBABLY true with Windows Terminal, but it is POSSIBLE that another terminal is sending keys slowly enough that when this line of code executes the next key simply hasn't arrived yet.

Yes you are right, I already faced that issue before, but in that case the code must be as the original. We really can't absolutely rely on the `Console.KeyAvailable` property but I am unaware how to deal with that issue.

Reply:
What I'm meaning here is when there is a Esc key resulting from a escape sequence, by a mouse event or F1 pressed, the `Console.KeyAvailable` must always be available, but sometimes it can return `false`, which will cause the escape sequence being corrupt.

Reply:
The right solution is to refactor things such that it's more like a protocol parser that buffers input and not rely on KeyAvailable. 

I have some ideas on this. I'll create a separate Issue where I explain further. 

For now, the current implementation is solid enough i think. But as we reduce our dependency on ncurses for Linux and switch WindowsDriver to use ansi sequencers we should make this more robust.